core/node/consensus/src/storage/vm_reader.rs

```diff
@@ -0,0 +1,294 @@
+        block_args: BlockArgs,
+    ) -> anyhow::Result<Vec<CommitteeValidator>> {
+        let mut committee = vec![];
+        let validator_committee_size = self
+            .read_validator_committee_size(block_args)
+            .await
+            .context("read_validator_committee_size()")?;
+        for i in 0..validator_committee_size {
+            let committee_validator = self
+                .read_committee_validator(block_args, i)
+                .await
+                .context("read_committee_validator()")?;
+            committee.push(committee_validator)
+        }
+        Ok(committee)
+    }
+
+    async fn read_attester_committee(
+        &self,
+        block_args: BlockArgs,
+    ) -> anyhow::Result<Vec<CommitteeAttester>> {
+        let mut committee = vec![];
+        let attester_committee_size = self
+            .read_attester_committee_size(block_args)
+            .await
+            .context("read_attester_committee_size()")?;
+        for i in 0..attester_committee_size {
+            let committee_validator = self
+                .read_committee_attester(block_args, i)
+                .await
+                .context("read_committee_attester()")?;
+            committee.push(committee_validator)
+        }
+        Ok(committee)
+    }
+
+    async fn read_validator_committee_size(&self, block_args: BlockArgs) -> anyhow::Result<usize> {
+        let func = self
+            .registry_contract
+            .function("validatorCommitteeSize")
+            .unwrap()
+            .clone();
+
+        let tx = self.gen_l2_call_tx(self.registry_address, func.short_signature().to_vec());
+
+        let res = self.eth_call(block_args, tx).await;
+
+        let tokens = func.decode_output(&res).context("decode_output()")?;
+        U256::from_tokens(tokens)
+            .context("U256::from_tokens()")
+            .map(|t| t.as_usize())
+    }
+
+    async fn read_attester_committee_size(&self, block_args: BlockArgs) -> anyhow::Result<usize> {
+        let func = self
+            .registry_contract
+            .function("attesterCommitteeSize")
+            .unwrap()
+            .clone();
+        let tx = self.gen_l2_call_tx(self.registry_address, func.short_signature().to_vec());
+
+        let res = self.eth_call(block_args, tx).await;
+        let tokens = func.decode_output(&res).context("decode_output()")?;
+        U256::from_tokens(tokens)
+            .context("U256::from_tokens()")
+            .map(|t| t.as_usize())
+    }
+
+    async fn read_committee_validator(
+        &self,
+        block_args: BlockArgs,
+        idx: usize,
+    ) -> anyhow::Result<CommitteeValidator> {
+        let func = self
+            .registry_contract
+            .function("validatorCommittee")
+            .unwrap()
+            .clone();
+        let tx = self.gen_l2_call_tx(
+            self.registry_address,
+            func.encode_input(&zksync_types::U256::from(idx).into_tokens())
+                .unwrap(),
```

Comment:
These unwraps could be at least `expect` with the method name to give us some context if they fail, in case we can't use bindings.